cmd/config.go

```diff
@@ -376,8 +410,10 @@ func prettyKeyForProject(k config.Key, proj *workspace.Project) string {
 // structure in the future, we should not change existing fields.
 type configValueJSON struct {
 	// When the value is encrypted and --show-secrets was not passed, the value will not be set.
-	Value  *string `json:"value,omitempty"`
-	Secret bool    `json:"secret"`
+	Value       *string     `json:"value,omitempty"`
+	Secret      bool        `json:"secret"`
+	Object      bool        `json:"object"`
+	ObjectValue interface{} `json:"objectValue,omitempty"`
```

Comment:
Does it make sense to put `ObjectValue` under `Value`, and group `Secret`, `Object`? It might make it easier to understand what's going on here. If I am guessing correctly, `Value` is used for the "non-object" case. And if the config value is a JSON object, then it will be stored in `ObjectValue`?

Did you consider just having `Value` be an `interface{}` and be either a `string` or `map`? It should for the most part "just work", except when marshaling a string TO JSON it will have `""` added to it, which may need to be removed later.

Reply:
> Does it make sense to put `ObjectValue` under `Value`, and group `Secret`, `Object`?

Sure.

> Did you consider just having `Value` be an `interface{}` and be either a `string` or `map`?

So, right now `Value` is always going to be a valid value. If `Object` is true, it'll be a JSON encoded string.

I didn't want to make `Value` be an actual object when `Object` is true, as that could break existing users who may be using this programmatically and always expecting Value to be a string.

I debated whether to add `ObjectValue` at all. It's not really necessary as you can get the JSON string from `Value` and parse it yourself if you want. `ObjectValue` more of just a convenience, to be able to just `jq` into object in `ObjectValue` rather than having to first retrieve the JSON string from `Value` and then parse that.

If `ObjectValue` adds more confusion and isn't useful in practice, I'd be fine with not including it.

Reply:
Do we need the `Object` flag at all? Could we just use a non-empty `ObjectValue` to detect an object?

Reply:
> Do we need the `Object` flag at all? Could we just use a non-empty `ObjectValue` to detect an object?

Good point. Removed the bool `Object` property.